### PR TITLE
Add default storage class

### PR DESCRIPTION
**Pull request by galal-hussein**
https://github.com/rancher/k3s/issues/85

### CODE REVIEW

File: manifests/local-path-storage.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        app: local-path-provisioner
+    spec:
+      serviceAccountName: local-path-provisioner-service-account
+      containers:
+      - name: local-path-provisioner
+        image: rancher/local-path-provisioner:v0.0.9
+        imagePullPolicy: IfNotPresent
+        command:
+        - local-path-provisioner
+        - --debug
```
Comment (cjellick) [inline manifests/local-path-storage.yaml]: should we turn this off?

File: manifests/local-path-storage.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: local-path-provisioner-role
+subjects:
+- kind: ServiceAccount
+  name: local-path-provisioner-service-account
+  namespace: local-path-storage
+---
+apiVersion: apps/v1beta2
```
Comment (cjellick) [inline manifests/local-path-storage.yaml]: go to latest version

File: pkg/cli/cmds/server.go [hunk 3]
```diff
@@ -189,6 +190,12 @@ func NewServerCommand(action func(*cli.Context) error) cli.Command {
 				Usage:       "Disable Kubernetes default scheduler",
 				Destination: &ServerConfig.DisableScheduler,
 			},
+			cli.StringFlag{
+				Name:        "default-local-storage-path",
+				Usage:       "Default local storage path for local provisioner storage class",
+				Destination: &ServerConfig.DefaultLocalStoragePath,
+				Value:       "/opt/local-path-provisioner",
```
Comment (ibuildthecloud) [inline pkg/cli/cmds/server.go]: `/var/lib/rancher/k3s/storage`

File: manifests/local-path-storage.yaml [hunk 4]
```diff
@@ -0,0 +1,109 @@
+apiVersion: v1
```
Comment (ibuildthecloud) [inline manifests/local-path-storage.yaml]: rename file to `local-storage`

File: manifests/local-path-storage.yaml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+kind: Namespace
+metadata:
+  name: local-path-storage
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: local-path-provisioner-service-account
+  namespace: local-path-storage
```
Comment (ibuildthecloud) [inline manifests/local-path-storage.yaml]: Move kube-system

File: manifests/local-path-storage.yaml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  verbs: ["create", "patch"]
+- apiGroups: ["storage.k8s.io"]
+  resources: ["storageclasses"]
+  verbs: ["get", "list", "watch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: local-path-provisioner-bind
+  namespace: local-path-storage
```
Comment (ibuildthecloud) [inline manifests/local-path-storage.yaml]: drop namespace

File: pkg/cli/cmds/server.go [hunk 7]
```diff
@@ -189,6 +190,12 @@ func NewServerCommand(action func(*cli.Context) error) cli.Command {
 				Usage:       "Disable Kubernetes default scheduler",
 				Destination: &ServerConfig.DisableScheduler,
 			},
+			cli.StringFlag{
+				Name:        "default-local-storage-path",
+				Usage:       "Default local storage path for local provisioner storage class",
+				Destination: &ServerConfig.DefaultLocalStoragePath,
+				Value:       "/var/lib/rancher/k3s/storage",
```
Comment (ibuildthecloud) [inline pkg/cli/cmds/server.go]: Don't set the Value

**Pull request merged**